pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -220,3 +224,53 @@ func draAdminAccessFeatureInUse(claim *resource.ResourceClaim) bool {
 
 	return false
 }
+
+func dropDisabledDRAResourceClaimDeviceStatusFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
+		newClaim.Status.Devices = nil
+	}
+}
+
+// dropDeallocatedStatusDevices removes the status.devices that were allocated
+// in the oldClaim and that have been removed in the newClaim.
+func dropDeallocatedStatusDevices(newClaim, oldClaim *resource.ResourceClaim) {
```

Comment:
commented in https://github.com/kubernetes/kubernetes/pull/128240/files#r1823538165